core/src/serve_repair.rs

```diff
@@ -455,20 +470,72 @@ impl ServeRepair {
         stats.dropped_requests_load_shed += dropped_requests;
         stats.total_requests += total_requests;
 
-        let root_bank = self.bank_forks.read().unwrap().root_bank();
-        let epoch_staked_nodes = root_bank.epoch_staked_nodes(root_bank.epoch());
-        for reqs in reqs_v {
-            self.handle_packets(
-                ping_cache,
-                recycler,
-                blockstore,
-                reqs,
-                response_sender,
-                stats,
-                data_budget,
-                &epoch_staked_nodes,
-            );
+        let decode_start = Instant::now();
+        let mut decoded_reqs = Vec::default();
+        for batch in reqs_v {
+            for packet in &batch {
```

Comment:
maybe instead of nested for loops and double indentation something like:
```rust
for packet in reqs_v.iter().flatten() {
```

Reply:
changed